src/xbot/common/subsystems/ConnectionMonitorSubsystem.java

```diff
@@ -0,0 +1,42 @@
+package xbot.common.subsystems;
+
+import com.google.inject.Singleton;
+import edu.wpi.first.wpilibj.Timer;
+import xbot.common.command.BaseSubsystem;
+import xbot.common.logic.Latch;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+import java.util.Observable;
+import java.util.Observer;
+
+@Singleton
+public class ConnectionMonitorSubsystem extends BaseSubsystem implements Observer {
+
+    protected final DoubleProperty timeOut;
+    private final Latch connectionLatch = new Latch(true, Latch.EdgeType.FallingEdge);
+
+    private double lastPacketReceivedTimestamp = Timer.getFPGATimestamp();
```

Comment:
This will mean that the monitor assumes it was connected whenever the system boots, which seems like the wrong assumption. I'd prefer setting this to `0` or making the field nullable and using `null`.

Reply:
Made it -1